### DISCUSSION TITLE
Azure Functions scaled by KEDA on virtual nodes

### DISCUSSION THREAD

**Discussion by christle** [kedacore/keda#1972] [Q&A / Need Help]
Hello,

i know, this is not a directly a KEDA topic, but i hope that i can find someone with the same setup in this community :)

We use azure-monitor for http, queue and eventhub trigger for nearly 80 azure functions. now i want to use a virtual node to get rid of vm startup times in scaling scenarios. But currently i run in a for me unsolvable situation:

all azure functions stuck in a restart loop. As i understand, the reason is, that azure functions send internally after every duration a "signal 0". The ACI interpret this and terminate the process. Then happens a restart. in ACI directly it is possible to set the restart policy to OnFailure, but over AKS and the virtual node connector, that is not possible. The only accepted value for the restart-policy in deployments are "Always".

In my mind are Azure Functions, KEDA on AKS + Virtual nodes a powerful way and i thought this is a common setup, but how to solve this issue?

Have anybody here experience with it?

**Comment by tomkerkhove** [comment]
I haven't seen this to be honest, maybe @jeffhollan or @ahmelsayed can help here.

As far as I know, if the host is up it will stay up. Can you elaborate on the "signal 0" please?

**Reply by christle** [reply]
we run into this issue: https://github.com/Azure/AKS/issues/1040. 

Like the first reply, we use the azure function docker image (for dotnet) in a way like the func cli produce it. With a normal cluster vm, all function hosts keep running. On ACI the function shutdown right after the start. Maybe the function start cmd inside the docker (base)image sending a success signal (status code 0 after execution), with initiates the shutdown. 

I will make some further investigations, but i hoped anyone else run in the same situation, because i think we don't use something special in our setup.

**Reply by tomkerkhove** [reply]
That's a good question, but maybe one to better ask on the Azure Function repo?

Feel free to circle back here if you know more though.

**Reply by christle** [reply]
sounds good. Thank you
